### ISSUE TITLE
JSONAPISerializer Cannot use include function

### ISSUE THREAD

**Issue by richmolj** [samselikoff/ember-cli-mirage#851]
From the docs:

```
You can also define include as a function so it can be determined dynamically:
```

```js
// mirage/serializers/author.js
export default Serializer.extend({
  include: function(request) {
    if (request.queryParams.posts) {
      return ['blogPosts'];
    } else {
      return [];
    }
  }
});
```

It looks like this does not work for JSONAPISerializer. The base serializer [will check if include is a function or a property](https://github.com/samselikoff/ember-cli-mirage/blob/e196721e6b9400509f8b004ce932afa697671df6/addon/serializer.js#L476-L485) but the JSONAPISerializer [always calls it like a property](https://github.com/samselikoff/ember-cli-mirage/blob/e196721e6b9400509f8b004ce932afa697671df6/addon/serializers/json-api-serializer.js#L250).

**Comment by samselikoff** [comment]
Yep, someone just mentioned this, definitely a bug. Should be fixed with #781, as it's basically a rewrite of serializer layer.

**Comment by samselikoff** [comment]
Duplicate of #815
